Оформление блока categories

nylus-ark opened the pull request:
Я доделал блок categories с карточками продуктов! Некоторые вопросы написал внутри css.

nylus-ark committed:
Исправляет оформление footer и начинает добавлять оформление для categories

nylus-ark committed:
Добавляет оформление categories

Так же вносит мелкие правки в footer

nylus-ark committed:
Добавляет оформление блока Delivery

Так же вносит правки по вложенности в css

nylus-ark committed:
Добавляет оформление блока Partners

redly commented on index.html:
```diff
@@ -50,24 +50,36 @@ <h2 class="title-h2 categories-title">
+                    <li class="apparel-list">
                         <a href="#">
-                            <img src="img/apparel.png" alt="Apparel Kids">
+                            <div class="img-apparel">
+                                <img src="img/apparel.png" alt="Apparel Kids">
+                                <h3 class="title-h3 apparel-title">Apparel Kids</h3>
+                            </div>
                         </a>
                     </li>
-                    <li>
+                    <li class="shop-list">
                         <a href="#">
-                            <img src="img/shop.png" alt="Holiday shop">
+                            <div class="img-shop">
+                                <img src="img/shop.png" alt="Holiday shop">
+                                <h3 class="title-h3 shop-title">Holiday shop</h3>
+                            </div>
                         </a>
                     </li>
```
По аналогии с тем, как ты исправил `delivery-text` нужно и тут внести изменения
Нет смысла делать разные классы на каждый li, они ничем не отличаются
Как и нет смысла делать разные классы на заголовки

Разные классы стоит задать на div, который у тебя будет отвечать за разные фоны
Но у тебя уже есть ссылка, которую можно использовать как обертку и не добавлять лишний div, так верстка станет более чистой

Итого, если переписать код одного li, может быть что-то такое

```html
<li class="products-list-item"> <!-– абстрактный класс у списка -->
    <a href="#" class="products-list-item-wrapper bg-green"> <!-– добавляем класс для обертки и отдельный класс для фона -->
        <img src="img/apparel.png" alt="Apparel Kids">
        <h3 class="title-h3">Apparel Kids</h3>
    </a>
</li>
```

redly reviewed (changes requested):
Пока писал ревью, ты часть подправил сам
В целом все хорошо, небольшое замечание и нужно двигаться дальше

redly commented on css/style.css:
```diff
@@ -139,127 +150,435 @@ a.header-logo:active,
+				/*
+				box-shadow: 0px 0px 40px 0px #930303 25%;
+				
+				Нужно ли это переносить из макета?
```
Нет, не нужно

redly commented on css/style.css:
```diff
@@ -139,127 +150,435 @@ a.header-logo:active,
+				height: 64px;
+				left: 26px;
+				top: 90px;
+			}
```
В целом этот подход к верстке очень плохой
Тебе приходится подгонять и изображение, и заголовок отдельно

Нужно
1. Сам блок, в котором заголовок и изображение сделать флексом, указать что все что внутри него должно быть по центру вертикально и горизонтально
2. Изображение положить абсолютом, но выравнить его просто по центру
Чтобы располагать блоки, неизвестного размера по центру стоит воспользоваться вот такой практикой
```css
position: absolute;
top: 50%;
left: 50%;
transform: translate(-50%, -50%);
```

То есть ты размещаешь его по центру, а потом через transform сдвигаешь на половину его ширины/высоты, потому что изначально центр будет рассчитываться от верхнего левого угла

Собственно эти два пункта помогут тебе не подгонять каждую категорию, а написать общий код для всех категорий

Твой подход плох тем, что
- При изменении размеров изображений придется менять CSS
- При изменении контента (текста / картинки) опять же придется менять и подгонять CSS

Эти пункты говорят о том, что код будет неподдерживаемым
Как следствие из них вытекает то, что подход этот очень долгий. Для каждой категории приходится писать много своего уникального кода

nylus-ark replied:
Я добавил изображениям position: relative, а заголовкам absolute и они отображаются по центру, нужно ли уточнять разметку через top left и transform, если все работает без этого?

redly commented on css/style.css:
```diff
@@ -78,54 +90,53 @@ body {
-	text-decoration: none;
-	text-transform: uppercase;
-}
+		.header-nav-list li a {
```
Не стоило повышать специфичность селектора
В твоем случае, что `header-nav-list a`, что `header-nav-list li a` делают одно и то же. Нет ссылок вне li, которые ты можешь "задеть" более общим селектором

А вот увеличенная специфичность сильно мешает поддержке кода

Старайся всегда использовать максимум уровень вложенности 2. 
Здесь это возможно, поэтому лучше вернуть как было

redly commented on css/style.css:
```diff
@@ -139,127 +150,435 @@ a.header-logo:active,
+		height: 270px;
+	}
+
+		.product-list li a {
```
Аналогично комментарию выше

`.product-list li a` -> `.product-list a`

redly commented on css/style.css:
```diff
@@ -139,127 +150,435 @@ a.header-logo:active,
+	justify-content: center;
+	align-items: flex-start;
+	width: 480px;
+	height: inherit;
```
height и так по умолчанию inherit
нет смысла указывать это дополнительно

redly replied:
Вот этот комментарий не отработал

redly commented on css/style.css:
```diff
@@ -139,127 +150,435 @@ a.header-logo:active,
+		width: 326px;
+	}
+
+		.delivery-list li a {
```
Аналогично комментариям выше про специфичность

nylus-ark committed:
Добавляет оформление блока Main-bottom

Так же вносит правки в блок Categories

redly commented on css/style.css:
```diff
@@ -139,127 +150,435 @@ a.header-logo:active,
+	display: flex;
+	flex-direction: row;
+	list-style: none;
+	height: 61px;
```
Ограничивая так жестко высоту ты не даешь возможности на переполнение этого блока
Например, если мы добавим больше логотипов верстка поедет

Также она поедет еще и поэтому, что по умолчанию flex-wrap: nowrap

Итого, нужно определиться, что делать. 
Варианта 2: 

1. Если мы хотим ограничить, чтобы логотипы были только в одну строчку и не больше чем в макете, то нужно хотя бы добавить `overflow: hidden`. Тогда при добавлении большего кол-ва логотипов они не будут выходить за границы справа

2. Если мы хотим, чтобы блок нормально себя вел при переполнении (предпочтительный вариант), то нужно
заменить height на min-height и сделать возможным переход на новую строку для логотипов, когда их больше 4х, то есть добавить flex-wrap: wrap

redly commented on css/style.css:
```diff
@@ -139,127 +150,435 @@ a.header-logo:active,
+	}
 
-}
+	.partners-list li:last-child {
```
Как следствие из пункта 2 выше, партнеров по идее может быть гораздо больше чем 4, поэтому last-child будет работать некорректно. Нам нужно сбрасывать margin-right для каждого 4-го пункта, то есть лучше использовать nth-child(4n)

Но стоит также попробовать сделать просто `justify-content: space-between` для `.partners-list li`. Скорее всего так все будет ок с отступами

nylus-ark replied:
До этого пробовал сделать space-between для partners-list, но ничего не работало. Оказалось надо просто указать ширину списка и все заработало без отступов (хотя я думал список наследует ширину контейнера).

redly commented on css/style.css:
```diff
@@ -139,127 +150,435 @@ a.header-logo:active,
+				margin: 25px 10px 0 0;
+			}
+
+				.footer-col-list li a {
```
Аналогично комментариям выше про специфичность

redly commented on css/style.css:
```diff
@@ -139,127 +150,435 @@ a.header-logo:active,
+			margin: 15px 0;
+		}
+
+			.footer-col-list li a {
```
Аналогично комментариям выше про специфичность

redly commented on css/style.css:
```diff
@@ -139,127 +150,438 @@ a.header-logo:active,
+		}
+
+		.products-list-item-wrapper h3 {
+			/*
+			Что бы отобразить слова в две строки, 
+			задал ширину по самому длинному загаловку.
+			Напиши есть ли какой то способ лучше.
+			*/
+			width: 192px; 
```
Все нормально тут, можно взять с запасом чуть ширину, например, `max-width: 200px`
И лучше заменить на max-width

nylus-ark committed:
Вносит правки в css

Убирает повышение специфичности селекторов и делает мелкие правки в блоках categories и partners

redly commented on css/style.css:
```diff
@@ -139,127 +150,431 @@ a.header-logo:active,
+
+.dlk-aside-container * {
+	margin: 0 0 10px 30px;
+}
```
Очень плохой подход, селектор звездочку лучше не использовать никогда
Слишком специфичный селектор, потом будет сложно менять эти стили

redly commented on css/style.css:
```diff
@@ -139,127 +150,431 @@ a.header-logo:active,
+
+.dlk-aside-container *:last-child {
+	margin-top: 10px;
 }
```
Аналогично комментарию выше

redly commented on css/style.css:
```diff
@@ -139,127 +150,431 @@ a.header-logo:active,
-	text-decoration: none;
+.contacts-aside-container *:last-child {
+	margin-bottom: 0;
 }
```
Ну и это тоже
Лучше задавать какие-то общие классы, так проще будет поддерживать код

redly commented on css/style.css:
```diff
@@ -139,127 +150,431 @@ a.header-logo:active,
+			line-height: 32px;
+			text-decoration: none;
+			height: 55px;
+			padding: 11px 180px 11px 60px;
```
Очень странные padding, обычно display блок должно было хватить, почему такой слева большой получился не ясно. Возможно можно через max-width / text-align: center решить вопрос, это будет лучше

Больше padding плохи тем, что при адаптивном дизайне придется их менять каждый раз, будет много лишнего кода

nylus-ark replied:
Убрал большой padding слева, верстке это не помешало. Если совсем убрать padding и задать max-width то сам блок с цветной подсветкой обрезается, а text-align у меня выравнивает текст ссылки по горизонтали, а мне нужно выравнивать по вертикали. В итоге я не стал пока такой подход оставлять.

nylus-ark committed:
Добавляет внутреннюю страницу Catalog

Так же вносит правки в блоки delivery,dlk и contacts

redly commented on catalog.html:
```diff
@@ -0,0 +1,228 @@
+                        </a>
+                    </li>
+                </ul>
+                <ul class="products-list"> 
```
Это должен быть единый список

Просто 4й элемент нужно смещать.
Чтобы такое смещение было только на странице каталога, стоит предусмотреть у списка доп. класс

redly commented on catalog.html:
```diff
@@ -0,0 +1,228 @@
+                    <a href="catalog.html">Catalog</a>
+                </li>
+            </ul>
+        </div>
```
Лучше было бы обернуть в container оба этих блока, чем дублировать класс
Также это похоже на шапку внутренних страниц

Для main я бы задал класс – `content`
Тогда эту шапку можно было бы разметить как `<div class="content-header container">`

В целом, можно было бы вообще после main сделать общий container для всего, потому что у тебя там весь контент выравнивается, но этот вариант не очень хорошо будет работать для страницы Toys (там фильтры на весь экран)

redly commented on catalog.html:
```diff
@@ -0,0 +1,228 @@
+
+        <section class="description">
+            <div class="container description-container">
+                <div class="description-container-wrapper">
```
Вполне достаточно было бы сделать 2 блока – section полноразмерный и внутри container которым выравнить текст. Зачем три вложенности не ясно. Чем меньше ты используешь элементов в разметке, тем лучше

redly commented on catalog.html:
```diff
@@ -0,0 +1,228 @@
+            </div>
+        </section>
+
+        <section class="description">
```
Название description слишком общее, может использоваться еще много где и стили могут пересечься, будут проблемы

Также блок выглядит как общий текстовый. В такие блоки обычно контент попадает из админки и важно, чтобы все стилизовалось через каскад

Можно задать класс такой `text-content` и остальная стилизация будет `.text-content h2`, `.text-content p`

redly commented on catalog.html:
```diff
@@ -0,0 +1,228 @@
+            <div class="container description-container">
+                <div class="description-container-wrapper">
+                    <h2 class="title-h2">Product Description</h2>
+                    <p class="description-container-text">
+                    Interactive doll Zapf Creation Baby Born Sister can drink from a bottle and 
+                    almost really cry. You can even bathe her, just like a little sister.
+                    </p>
+                    <p class="description-container-text">
+                    A toy with a height of 43 cm bends its knees, the head rotates, it can stand 
+                    upright. Such a doll will be an excellent choice for a gift for a girl from 4 
+                    years old.
+                    </p>
+                    <p class="description-container-text">
+                    The set includes a miniature set of fashionable clothes, a hairbrush, a bottle 
+                    for water, a pair of hairpins and elastic bands for hair, as well as two bracelets: 
+                    one for the doll, the other for her young mistress.
+                    </p>
```
Как описал выше стоит избегать доп классов в подобных текстовых блоках

redly commented on css/catalog.css:
```diff
@@ -0,0 +1,446 @@
+*::before,
+*::after {
+	box-sizing: inherit;
+}
```
Мы неправильно друг друга поняли

Сейчас ты полностью дублируешь стили, которые общие для body, header, footer и тп
Это делать ни в коем случае не стоит

Варианта 2: 
1. Вынести общие стили в common.css. Для каждой новой страницы делать свою таблицу стилей, по типу catalog.css, toys.css
2. Писать все стили в одном файле style.css

Обычно при написании стилей без препроцессора используют второй вариант
Когда есть сборка и более продвинутые инструменты работы используют первый вариант, но там он чуть-чуть трансформируется

Использовать одну таблицу стилей лучше с точки зрения производительности – меньше запросов на сервер, меньше ожидание и как следствие быстрее отрисовка

nylus-ark committed:
Вносит правки в Catalog

redly commented on catalog.html:
```diff
@@ -0,0 +1,224 @@
+        </section>
+
+        <section class="text-content">
+                <div class="container text-content-container">
+                    <h2 class="title-h2">Product Description</h2>
+                    <p>
+                    Interactive doll Zapf Creation Baby Born Sister can drink from a bottle and 
+                    almost really cry. You can even bathe her, just like a little sister.
+                    </p>
+                    <p>
+                    A toy with a height of 43 cm bends its knees, the head rotates, it can stand 
+                    upright. Such a doll will be an excellent choice for a gift for a girl from 4 
+                    years old.
+                    </p>
+                    <p>
+                    The set includes a miniature set of fashionable clothes, a hairbrush, a bottle 
+                    for water, a pair of hairpins and elastic bands for hair, as well as two bracelets: 
+                    one for the doll, the other for her young mistress.
+                    </p>
+                </div>
```
Поправь тут отступы
Лишние пробелы слева у `<div class="container text-content-container">` и недостаточно пробелов у текста внутри `p`

redly reviewed (approved):
Всё хорошо, за исключением отступов

PR принимаю, а то слишком много изменений уже
Давай дальше опять двигаться мелкими блоками
